students/ghassan/lesson04/trig.py

```diff
@@ -10,8 +10,20 @@
 
 for item in range(0, len(data)-2):
     z.setdefault('{} {}'.format(data[item], data[item+1]), [])
-    z['{} {}'.format(data[item], data[item+1])].append('{}'.format(data[item+2]))
+    z['{} {}'.format(
+        data[item], data[item+1])].append('{}'.format(data[item+2]))
 
```

Comment:
printing `z` here would produce:
```
{'I  ': ['w', 'm', 'w', 'm'], '  w': ['i', 'i'], 'w i': ['s', 's'], 'i s': ['h', 'h'], 's h': [' ', ' '], 'h  ': ['I', 'I'], '  I': [' ', ' ', ' '], '  m': ['a', 'i'], 'm a': ['y'], 'a y': [' '], 'y  ': ['I'], 'm i': ['g'], 'i g': ['h'], 'g h': ['t']}```
is that intentional? what is the purpose of that?

Reply:
well, for this part, the data is a list: `data = "I wish I may I wish I might".split()`
You're right though, the second section where it writes to a file is not working as I intended.

Reply:
I updated the second part. I think it should work as intended now
`(venv) galmamar (master *) lesson04 $ cat new_book.txt
I may I wish I I`